sklearn/tree/_criterion.pyx

```diff
@@ -1103,11 +1112,14 @@ cdef class MAE(RegressionCriterion):
                 # remove everything from left and put it into right
                 (<WeightedMedianCalculator> left_child[k]).pop(&value,
                                                                &weight)
+                # push method ends up calling safe_realloc, hence `except -1`
                 (<WeightedMedianCalculator> right_child[k]).push(value,
                                                                  weight)
+        return 0
 
-    cdef void reverse_reset(self) nogil:
-        """Reset the criterion at pos=end."""
+    cdef int reverse_reset(self) nogil except -1:
+        """Reset the criterion at pos=end. Returns -1 upon error and 0 when
+           succeeded."""
```

Comment:
rather than just "error" and "success" it would be more informative to say: "Return -1 in case of failure to alocate memory and 0 otherwise."